tests/common/pkt_filter/filter_pkt_in_buffer.py

```diff
@@ -0,0 +1,232 @@
+from collections import OrderedDict
+
+import time
+import json
+import copy
+
+import ptf.mask as mask
+import ptf.packet as packet
+
+
+NATIVE_TYPE = (int, float, long, bool, list, dict, tuple, set, str, bytes, unicode, type(None))
+
+
+def _parse_layer(layer):
+    """
+    Convert packet layer to dictionary
+
+    Args:
+        layer: Layer of packet
+
+    Returns:
+        Field dictionary
+    """
+    fields = {}
+
+    if not hasattr(layer, 'fields_desc'):
+        return None
+
+    for field in layer.fields_desc:
+        value = getattr(layer, field.name)
+        if value is type(None):
+            value = None
+
+        if not isinstance(value, NATIVE_TYPE):
+            value = _parse_layer(value)
+
+        fields[field.name] = str(value)
+
+    return {layer.name: fields}
+
+
+def convert_pkt_to_dict(pkt):
+    """
+    Convert scapy packet to dictionary
+
+    Args:
+        pkt: Scapy packet
+
+    Returns:
+        Packet dictionary
+    """
+    packet_dict = OrderedDict()
+    counter = 0
+
+    while True:
+        layer = pkt.getlayer(counter)
+        if not layer:
+            break
+
+        layer_dict = _parse_layer(layer)
+        if layer_dict:
+            packet_dict.update(layer_dict)
+
+        counter += 1
+
+    return packet_dict
+
+
+class FilterPktBuffer(object):
+    """
+    FilterPktBuffer class for finding of packets in the buffer of PTF
+    """
+    def __init__(self, ptfadapter, exp_pkt, dst_port_number, match_fields=None, ignore_fields=None):
+        """
+        Initialize an object for finding packets in the buffer
+
+        Args:
+            ptfadapter: PTF adapter
+            exp_pkt: Expected packet
+            dst_port_number: Destination port number
+            match_fields: List of packet fields that should be matched
+            ignore_fields: List of packet fields that should be ignored
+        """
+        self.received_pkt = None
+        self.received_pkt_diff = []
+        self.ptfadapter = ptfadapter
+        self.pkt = exp_pkt
+        self.dst_port_number = dst_port_number
+
+        if match_fields is None:
+            match_fields = []
+        self.match_fields = match_fields
+
+        if ignore_fields is None:
+            ignore_fields = []
+        self.ignore_fields = ignore_fields
+
+        self.masked_exp_pkt = mask.Mask(self.pkt)
+        self.pkt_dict = convert_pkt_to_dict(self.pkt)
+
+
+    def __ignore_fields(self):
+        """
+        Ignore fields of packet
+        """
+        for field, value in self.ignore_fields:
+            self.masked_exp_pkt.set_do_not_care_scapy(getattr(packet, field), value)
```

Comment:
Typo with the `packet` variable?

Reply:
Could you please specify the line of code with the typo?

Reply:
I see, `packet` here is an imported module, not some local undefined `variable`.